### PR TITLE
Hide Search build system from DTK

### PR DESCRIPTION
**Pull request by aprokop**

### CODE REVIEW

**Comment by sslattery** [comment]
How does moving the search build up from the subdirectory hide it?

**Comment by aprokop** [comment]
It does not hide the Search itself, just its build system. This way, DTK won't touch any CMakeLists.txt in the Search directory, and thus @dalg24 can put in the independent system in prior to the move out.

**Comment by sslattery** [comment]
ok so this is making seem as though the search lib is coming in externally, even though it is not, and therefore we can change to that model later on more easily

**Comment by aprokop** [comment]
Exactly. The plan is to have ArborX be a submodule or a snapshot within DTK. We can also discuss having ArborX as a proper dependency later.

File: packages/Search/CMakeLists.txt [hunk 1]
```diff
@@ -1,6 +1,18 @@
...
+find_package(kokkos REQUIRED)
+find_package(Boost REQUIRED COMPONENTS program_options unit_test_framework)
+find_package(MPI REQUIRED)
 
-TRIBITS_ADD_EXAMPLE_DIRECTORIES(examples)
-TRIBITS_ADD_TEST_DIRECTORIES(test)
+add_library(ArborX INTERFACE)
+target_link_libraries(ArborX INTERFACE Kokkos::kokkos MPI::MPI_CXX)
+#target_compile_features(ArborX INTERFACE cxx_std_14)
+target_link_directories(ArborX INTERFACE /usr/local/cuda/lib64)
```
Comment (dalg24) [inline packages/Search/CMakeLists.txt]: oops

File: packages/Search/CMakeLists.txt [hunk 2]
```diff
@@ -1,6 +1,18 @@
-ADD_SUBDIRECTORY(src)
+cmake_minimum_required(VERSION 3.13)
```
Comment (aprokop) [inline packages/Search/CMakeLists.txt]: Meh, I would want to relax this later.

File: packages/Search/CMakeLists.txt [hunk 3]
```diff
@@ -1,6 +1,18 @@
...
+project(ArborX CXX)
 
-TRIBITS_INCLUDE_DIRECTORIES(examples/point_clouds)
+find_package(kokkos REQUIRED)
+find_package(Boost REQUIRED COMPONENTS program_options unit_test_framework)
```
Comment (aprokop) [inline packages/Search/CMakeLists.txt]: Add `geometry`.

File: packages/Search/examples/bvh_driver/CMakeLists.txt [hunk 4]
```diff
@@ -1,44 +1,9 @@
...
-    bvh
-    POSTFIX_AND_ARGS_0 openmp --benchmark_filter=OpenMP --benchmark_color=false
-    COMM serial mpi
-    NUM_MPI_PROCS 1
-    NUM_TOTAL_CORES_USED 4
-    ENVIRONMENT OMP_NUM_THREADS=4
-    FAIL_REGULAR_EXPRESSION "data race;leak;runtime error"
-    )
-ENDIF()
+add_executable(ArborX_BoundingVolumeHierarchy.exe bvh_driver.cpp)
```
Comment (aprokop) [inline packages/Search/examples/bvh_driver/CMakeLists.txt]: Please change the name back to `ArborX_BVH.exe`.

**Comment by aprokop** [comment]
I cannot actually test this PR due to having to patch Kokkos...

File: packages/Search/CMakeLists.txt (hunk 2 above)
Comment (dalg24) [reply packages/Search/CMakeLists.txt]: Not gonna happen :)

File: packages/Search/CMakeLists.txt (hunk 3 above)
Comment (dalg24) [reply packages/Search/CMakeLists.txt]: You only list boost libraries that were built, not header-only packages.

**Comment by aprokop** [comment]
@dalg24 Please rebase on top of search examples to include all the tests in Search/tests.

**Comment by dalg24** [comment]
restest this please

**Pull request merged**